neo/Consensus/ConsensusService.cs

```diff
@@ -153,12 +157,23 @@ private void Log(string message, LogLevel level = LogLevel.Info)
 
         private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
         {
-            if (context.State.HasFlag(ConsensusState.CommitSent))
+            if (message.NewViewNumber < context.ViewNumber)
+            {
+                // If we are at a higher view, we can send the regeneration msg.
+                Log($"send recovery from view: {message.ViewNumber} to view: {context.ViewNumber}");
+                localNode.Tell(new LocalNode.SendDirectly { Inventory = context.MakeRecoveryMessage() });
+                // Note: In the future, we may want to limit how many nodes will send a regeneration msg
```

Comment:
I think a good strategy here instead of having all consensus nodes respond, is to have the last 3 nodes that produced blocks respond. I can add a check here to see if this node is one of the last 3 validators.

@erikzhang What do you think of this strategy?

Reply:
```
n = 3f + 1
```

If a node is requesting a `RecoveryMessage`, then only `f` nodes need to respond, because at least one of the `f` nodes is a normal node, otherwise the premise of the fault node <=f is violated.

Reply:
@erikzhang I agree that would be ideal, but how do the nodes know which should respond? They won’t know which nodes are crashed and which are running.

I have an idea. The requester’s (sensor is the `ChangeView`’s validator index can be used to know who should respond. Let the requested validator index be `i` Then I can have the nodes who’s index is `i + x`, respond where `x` is the set from `(i+1)%n` to `(i+f)%n` be eligible to send the recovery message.

Reply:
@erikzhang Do you like this proposal? Or what criteria would you suggest for a node to decide whether it should respond?

Reply:
It's a good idea.

Reply:
I like this idea but it breaks the assumption of `f` byzantine nodes.
I think it should not be used for now.

Reply:
If this type of logic starts to be used it becomes even harder to design the cases of possible problems, because only a single byzantine node would be able to destroy our logic.

Reply:
That is right, considering that the node's payload is carrying all necessary information for the recovery.

Reply:
Thanks. I made changes to implement it.